sympy/diffgeom/diffgeom.py

```diff
@@ -398,26 +398,28 @@ def transformation(self, sys):
         elif key in self.relations:
             expr = Matrix(self.relations[key])
         elif key[::-1] in self.relations:
-            expr = Matrix(self._inverse_transformations(sys, self))
+            expr = Matrix(self._inverse_transformation(sys, self))
         else:
-            expr = Matrix(self._indirect_transformations(self, sys))
+            expr = Matrix(self._indirect_transformation(self, sys))
         return Lambda(signature, expr)
 
     @staticmethod
     def _inverse_transformation(sys1, sys2):
         # Find the transformation relation from sys2 to sys1
-        forward_transform = sys1.transform(sys2)
-        forward_syms, forward_results = forward_transform.args
+        forward_transform_expressions = sys1.transform(sys2)
+        # Coordinate transformations are defined in terms of Symbols rather than CoordSymbols, so must be converted
+        forward_transform_expressions = to_coord_symbol(sys1.symbols, forward_transform_expressions)
```

Comment:
`sys1.transform(sys2)` already consists of `CoordinateSymbol` so we don't need `to_coord_symbol()` function. (Removing this line showed no different result)

Reply:
Ah thanks, I hadn't realised that the `transform` method was doing this for me